ada.html.markdown

```diff
@@ -0,0 +1,395 @@
+procedure Empty is
+   --  This is a declarative part.
+begin
+   --  Statements go here.
+   null;  --  Do nothing here.
+end Empty;
+
+--  Ada compilers accept compilation units which can be library packages,
+--  tasks, sub-programs, generics, etc.
+
+--  This is where "context clauses" go, these can be pragmas or ```with```
+--  statements. ```with``` is equivalent to "include" or "import" in other
+--  languages.
+with Ada.Text_IO;  --  Get access to a library package.
+
+procedure Hello is
+begin
+   Ada.Text_IO.Put_Line ("Hello, world");
+
+   Ada.Text_IO.Put ("Hello again, world");
+   Ada.Text_IO.New_Line;
+end Hello;
+
+
+--  Ada has a real module system. Modules are called packages and are split into
+--  two component parts, the specification (this file) and a body, the other.
+--  It is important to introduce packages early, as you will be using them from
+--  the start.
+package Stuff is
+   --  We could add the following line in order to tell the compiler that this
+   --  package does not have to run any code before the "main" procedure starts.
+   --  pragma Preelaborate;
+
+   --  Packages can be nested within the same file or externally.
+   --  Nested packages are accessed via dot notation, e.g. Stuff.Things.My.
+   package Things is
+      My : constant Integer := 100;
+   end Things;
+
+   --  If there are sub-programs declared within the specification, the body
+   --  of the sub-program must be declared within the package body.
+   procedure Do_Something;  --  If a subprogram takes no parameters, it is not
+                            --  shown.
+
+   --  We can also make generic sub-programs.
+   generic
+      type Element is (<>);
+   procedure Swap (Left, Right : in out Element);
+
+   --  Sometimes we want to hide how a type is defined from the outside world
+   --  so that nobody can mess with it directly. The full type must be defined
+   --  within the private section below.
+   type Blobs is private;
+private
+   type Blobs is new Integer range -25 .. 25;
+end Stuff;
+
+
+package body Stuff is
+   --  Sub-program body.
+   procedure Do_Something is
+      --  We can nest sub-programs too.
+      --  Parameters are defined with the direction of travel, in, in out, out.
+      --  If the direction of travel is not specified, they are in by default.
+      function Times_4 (Value : in Integer) return Integer is
+      begin
+         return Value * 4;
+      end Times_4;
+
+      I : Integer := 4;
+   begin
+      I := Times_4 (I);
+   end Do_Something;
+
+
+   --  Generic procedure body.
+   procedure Swap (Left, Right : in out Element) is
+      Temp : Element := Left;
+   begin
+      Left  := Right;
+      Right := Temp;
+   end Swap;
+begin
+   --  If we need to initialise something within the package, we can do it
+   --  here.
+   Do_Something;
+end Stuff;
+
+
+with Ada.Unchecked_Conversion;
+with Ada.Text_IO;
+with Stuff;
+
+procedure LearnAdaInY is
+   --  Indentation is 3 spaces.
```

Comment:
Typical indentation is 3 spaces
Conventional? Common?

Reply:
Convention. Style Guide.